Sources/MongoSwift/MongoCollection+BulkWrite.swift

```diff
@@ -62,8 +62,7 @@ extension MongoCollection {
          *   - `EncodingError` if an error occurs while encoding the options to BSON.
          */
         public func addToBulkWrite(bulk: BulkWriteOperation, index: Int) throws {
-            let opts = try BSONEncoder().encode(self.options)
-
+            let opts = try bulk.encoder.encode(self.options)
```

Comment:
what's the benefit of having the bulk operation have its own coders? Is it a `CodingStrategyProvider`?

Reply:
We need access to the collection's encoder in `addToBulkWrite`, but `BulkWriteOperation` stores a `mongoc_collection_t` rather than the entire collection itself, so it can't access it. Alternatively, we could store a `MongoCollection` on the operation, or pass in the encoder as a parameter to `addToBulkWrite` instead.

Reply:
I think it should just store the `MongoCollection`. this would also reduce some duplication, for ex. `BulkWrite.isAcknowledged` can just directly call the `MongoCollection`'s `isAcknowledged` method.

Reply:
I now remember why I did it this way. So the `WriteModel`s that we define are actually internal structs of `MongoCollection`. So, `InsertOneModel` is actually a `MongoCollection<MyType>.InsertOneModel`. `BulkWriteOperation`, however, is just a simple wrapper around a `mongoc_bulk_operation_t` with no generic parameters. Because of this, I think in order to store a `MongoCollection<CollectionType>` here, we'd also have to parameterize `BulkWriteOperation`, which I think isn't within scope for this work. Also, as a side note, a user could pass in a specific write concern to the bulk write operation, so we wouldn't be able to use `MongoCollection`'s `isAcknowledged`.

Reply:
gotcha. it seems more reasonable to me to just pass it into the bulk operation as you do now one time, rather than doing a bunch of copies passing it in for every operation we add.